__tests__/routes/cube/download.test.js

```diff
@@ -0,0 +1,127 @@
+const router = require('../../../routes/cube/download');
+const request = require('supertest');
+const express = require('express');
+const app = express();
+const dbSetup = require('../../helpers/dbTestSetup');
+
+const Cube = require('../../../models/cube');
+const cubefixture = require('../../../fixtures/examplecube');
+const { buildIdQuery } = require('../../../serverjs/cubefn');
+const carddb = require('../../../serverjs/cards');
+
+const fixturesPath = 'fixtures';
+const exampleCube = cubefixture.exampleCube;
+const cubeID = exampleCube.shortID;
+
+app.use('/', router);
+
+let mongoServer;
+
+beforeAll(async () => {
+  mongoServer = await dbSetup.connect();
+  await carddb.initializeCardDb(fixturesPath, true);
+  await new Cube(exampleCube).save();
+});
+
+afterAll(async () => {
+  await dbSetup.close(mongoServer);
+  carddb.unloadCardDb();
+});
+
+test('cubecobra text download', () => {
+  return request(app)
+    .get('/cubecobra/' + cubeID)
+    .expect(200)
+    .expect('Content-Type', 'text/plain')
+    .expect('Content-disposition', 'attachment; filename=' + exampleCube.name + '.txt')
+    .expect((res) => {
+      const lines = splitText(res.text);
+      expect(lines[0]).toEqual('Acclaimed Contender [eld-1]');
+      expect(lines.length).toEqual(exampleCube.cards.length);
+    });
+});
+
+test('plaintext download', () => {
+  return request(app)
+    .get('/plaintext/' + cubeID)
+    .expect(200)
+    .expect('Content-Type', 'text/plain')
+    .expect('Content-disposition', 'attachment; filename=' + exampleCube.name + '.txt')
+    .expect((res) => {
+      const lines = splitText(res.text);
+      expect(lines[0]).toEqual('Acclaimed Contender');
+      expect(lines.length).toEqual(exampleCube.cards.length);
+    });
+});
+
+test('MTGO download', () => {
+  return request(app)
+    .get('/mtgo/' + cubeID)
+    .expect(200)
+    .expect('Content-Type', 'text/plain')
+    .expect('Content-disposition', 'attachment; filename=' + exampleCube.name + '.txt')
+    .expect((res) => {
+      const lines = splitText(res.text);
+      expect(lines[0]).toEqual('1 Acclaimed Contender');
+      expect(lines[1]).toEqual('2 Brazen Borrower');
+      // The two Brazen Borrowers in the cube are deduped
+      expect(lines.length).toEqual(exampleCube.cards.length - 1);
+    });
+});
+
+test('csv download', () => {
+  const headerLine =
+    'Name,CMC,Type,Color,Set,Collector Number,Rarity,Color Category,Status,Finish,Maybeboard,Image URL,Image Back URL,Tags,Notes,MTGO ID';
```

Comment:
I think the additional validations are a good idea and I'll look into them, but is the "fragility" here actually a bad thing? I see it as the most straightforward way to test the contract of the CSV download API. Any changes to the columns here is a change to that contract and I don't think those should happen willy-nilly. Do those changes happen more frequently than I'm imagining?

Reply:
I added an assertion that the output parses correctly as a CSV. In the process I discovered that the current code actually includes an extra comma at the end of the line, which papaparse can still accept since it's tolerant, but it does generate an error. I went ahead and removed that extra comma from writeCard, which I think is safe since it doesn't seem to be used anywhere else, but LMK if we need to keep it as-is for some reason.

IMO this set of assertions - valid CSV format, header line and first line are equal to hardcoded expectation - is where I want this test to be, to balance test simplicity with comprehensiveness.

There should also be a "round-trip" test for each format that ensures that downloading and then importing works correctly, but these will be a little more involved so I'd like to not tackle them in this PR, if that's OK.

Reply:
The contract for our csv is very specifically not dependent on the order of columns and accepting of adding and removing(to an extent) columns.

Reply:
I buy that it's not dependent on the order and I can update the test accordingly, but it seems like it's still fair to have a test update when you add or remove a column. Is that ok?

As someone who consumes your API for a different use case than re-importing into Cubecobra, I'd like to think there's some degree of stability on the data format.